### PR TITLE
Makes new surgeries check for tool qualities 

### PR DESCRIPTION
**Pull request by ACCount12**
Surgeries from #35520 now check tool types instead of tool paths. 

This PR also converts some surgery-related `attack_by` procs to `tool_act`.

Tag: code improvement.

### CODE REVIEW

File: code/modules/surgery/organs/autosurgeon.dm [hunk 1]
```diff
@@ -53,19 +53,23 @@
...
-				desc = "[initial(desc)] Looks like it's been used up."
+	else
+		return ..()
+
+/obj/item/device/autosurgeon/screwdriver_act(mob/living/user, obj/item/I)
+	if(!storedorgan)
+		to_chat(user, "<span class='notice'>There's no implant in [src] for you to remove.</span>")
+	else
+		var/turf/open/floorloc = get_turf(user)
+		floorloc.contents += contents
```
Comment (Cyberboss) [inline code/modules/surgery/organs/autosurgeon.dm]:
```dm
var/atom/drop_loc = user.drop_location()
for(var/I in src)
    var/atom/movable/AM = I
    AM.forceMove(drop_loc)
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by ACCount12** [comment]
@Cyberboss done

**Comment by ShizCalev** [comment]
doesn't compile.

**Review by Cyberboss — APPROVED** [review]

**Comment by ACCount12** [comment]
@Cyberboss thanks for fixing it for me

**Pull request merged**